components/org.wso2.carbon.identity.oauth.uma.grant/src/main/java/org/wso2/carbon/identity/oauth/uma/grant/UMA2GrantHandler.java

```diff
@@ -106,10 +107,19 @@ public boolean validateGrant(OAuthTokenReqMessageContext tokReqMsgCtx) throws Id
             throw new IdentityOAuth2Exception("Empty id-token.");
         }
 
-        String subject = getSubjectFromIDToken(idToken, tokReqMsgCtx.getOauth2AccessTokenReqDTO().getTenantDomain());
+        String tenantDomain = tokReqMsgCtx.getOauth2AccessTokenReqDTO().getTenantDomain();
+        String subject = getSubjectFromIDToken(idToken, tenantDomain);
+        boolean isValidTicket;
 
-        // Validate the permission ticket against the subject.
-        if (validatePermissionTicket(permissionTicket, subject)) {
+        try {
+            FrameworkUtils.startTenantFlow(tenantDomain);
```

Comment:
Start the tenant flow just before the try { block. 
Otherwise, if startTenantFlow() throws an error, the finally block gets executed trying to end a tenant flow that was not started

Reply:
fixed